### PR TITLE
Add banner and modal for user onboarding

### PR DESCRIPTION
**Pull request by zoek1**
##### Description
Implement the banner and modal for user onboarding on Figma

DEMO:
https://www.loom.com/share/8d484cd6ecb142cf8ed49d2ba493de14
##### Testing
image
DeepinScreenshot_20200518211827

### CODE REVIEW

**Comment by PixelantDesign** [comment]
looking good! Could hide the What is gitcoin banner @zoek1 ?

**Comment by frankchen07** [comment]
One of the tradeoffs here is that it may take attention away from Townsquare. 

The ideal outcome for us is that we get additional data as to what people are doing on our site, which already sort of have through our google analytic views of 2020. I've done a cut of these below:

Screen Shot 2020-04-20 at 12 56 39

Right off the bat we see that the major pathways are the homepage (townsquare), grants, and the explorer. You can filter by whatever view you want, but the landing page is either townsquare, grants, or the explorer, the starting page (session start) alternates between townsquare, grants, issues, and the explorer. Further steps shows the same type of behavior.

What I would like to see more of is hackathon pages represented in these steps.

If we breakdown by page before and page after:

for gitcoin.co:
Screen Shot 2020-04-20 at 13 58 30

We can see that a large percentage of unique paths is still from homepage to explorer, with grants, products, and landing coming in 2nd, 3rd, and 4th. It's pretty uniform from the page before -> gitcoin.co (meaning people navigate back to gitcoin.co from alot of different areas).

for gitcoin.co/grants:
Screen Shot 2020-04-20 at 13 58 53

The majority of ways grants is discovered is from the the homepage as well. Once they get into grants, they're pretty much following up with all links within the grants product.

for gitcoin.co/hackathon-list/
Screen Shot 2020-04-20 at 13 59 16

Similar to grants, the homepage is where people navigate to the hackathon list. Once they reach the hackathon list, they've entered the hackathon product, and generally they remain there.

The summary from this:

1. it tells me that the townsquare / homepage is still the landing and starting point for breakoffs into the product, and the paths to grants definitely exists
2. if anything, I would want to focus my efforts on exposing the hackathon much more and directing volume towards that
3. for both hackathon and grants, once they get in, that almost becomes a different type of "flow" and we want to ensure that there's a good sub-"onboarding" experience there.

This makes me question the purpose of this modal, since we can generally track where people are going, do we need to explicitly ask them or do we just ensure that each integrated experience has clear action items once they get in, and we make sure to naturally expose it to them on townsquare?

**Comment by zoek1** [comment]
Sure @PixelantDesign :)

**Comment by zoek1** [comment]
@thelostone-mc another for review :pray:

**Comment by PixelantDesign** [comment]
@zoek1 it looks like we are missing the fourth link to townsquare ono the modal.

**Comment by PixelantDesign** [comment]
@frankchen07 the purpose of the modal is not to see where people go....
the goal is this ---> ensure that users have clear action items once they get in.

**Comment by thelostone-mc** [comment]
@PixelantDesign / @frankchen07 could you give discuss this and tell us how to proceed !
Hit the the approve PR & we can get it in

**Comment by zoek1** [comment]
I added the missed button

**Comment by PixelantDesign** [comment]
@zoek1 could we dynamically pull in the number of devs on the platform from this page? it’s around 40k, scrill to bottom to see.

https://gitcoin.co/results

**Comment by zoek1** [comment]
Covered @pixalantdesign, i added the code to show dynamically the number of devs

**Comment by PixelantDesign** [comment]
could you provide screenshots @zoek1 ?

**Comment by zoek1** [comment]
I updated the screenshots in the description, you can see the new logos in the modal

**Comment by PixelantDesign** [comment]
Looks great! thanks!

**Comment by owocki** [comment]
LGTM @zoek1 ; just moved it to the @gitcoinco/engineers reivew board

File: app/townsquare/views.py [hunk 1]
```diff
@@ -332,6 +332,11 @@ def get_following_tribes(request):
 
 
 def town_square(request):
+    data_results = JSONStore.objects.filter(view='results', key=None)
+    audience = '39102'
```
Comment (owocki) [inline app/townsquare/views.py:336]: dont you want to get the `.first()` result?

also would appreciate if this was cached in redis or similar so were not hitting the DB with this query every tTS pageview

Comment (zoek1) [reply app/townsquare/views.py]: Done :)

File: app/townsquare/templates/townsquare/index.html [hunk 2]
```diff
@@ -89,7 +75,19 @@
               </ul>
             </div>
           </div>
-          <div class="row mt-4 townsquare_header">
+          <div class="row {% if user.user.is_authenticated %} mt-4 {% else %} mt-2 {% endif %}townsquare_header">
```
Comment (danlipert) [inline app/townsquare/templates/townsquare/index.html:78]: typo here: `user.user.is_authenticated` -> `user.is_authenticated`. Does this still work somehow?

File: app/townsquare/views.py [hunk 3]
```diff
@@ -332,6 +332,18 @@ def get_following_tribes(request):
 
 
 def town_square(request):
+    from app.redis_service import RedisService
+    redis = RedisService().redis
```
Comment (danlipert) [inline app/townsquare/views.py:336]: The redis instance needs to be instantiated globally - connecting it here on every request is going to cause performance issues in production.

Comment (zoek1) [reply app/townsquare/views.py]: I created one instance to module level and modify the creation of Redis instances, now it uses a singleton pattern.

File: app/townsquare/templates/townsquare/index.html (hunk 2 above)
Comment (zoek1) [reply app/townsquare/templates/townsquare/index.html]: No, it doesn't display the correct margin, my bad. :sweat_smile: Thanks!

**Comment by owocki** [comment]
pretty keen to get this out soon.  way better than existing page

**Comment by zoek1** [comment]
> Hi @zoek1 looking great!
> 
> Could we use the hackathon logo on this button?
> 
> Screen Shot 2020-05-17 at 10 19 33 PM

@PixelantDesign I forgot to update the screenshot, the logos match on each section.

**Comment by PixelantDesign** [comment]
@zoek1 could you share the screenshot of the logo matching? The last screenshot has the wrong logo.

@octavioamu how does this link up with the web3 connect screen that you're working on?
Could you also help connect @zoek1  with the skills flow when user selects hackathon? 

@zoek1 could we have a video of this experience? Thanks!

**Comment by zoek1** [comment]
@PixelantDesign the update demo https://www.loom.com/share/8d484cd6ecb142cf8ed49d2ba493de14

**Comment by PixelantDesign** [comment]
Thanks for sharing the video @zoek1! I noticed the wallet screen keeps popping up. @octavioamu let's sync on the best way to handle this. It could be frustrating for the user.

**Comment by octavioamu** [comment]
image
Seems this link have some color problems here.

**Comment by zoek1** [comment]
FIxed on 99c0524 @octavioamu

**Pull request merged**